### PR TITLE
19361 - Add refund status to routing slip

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:* https://github.com/bcgov/entity/issues/19361

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ViewRoutingSlip/RefundRequestForm.vue [hunk 1]
```diff
@@ -30,6 +39,57 @@
...
+              class="hover-btn ml-2"
+              @click="expendStatus"
+            >
+              Update Status
+              <v-icon dense>{{ isExpanded ? 'mdi-menu-up' : 'mdi-menu-down' }}</v-icon>
+            </v-btn>
+          </template>
+          <v-list>
+            <v-list-item
+                v-for="status in RoutingSlipRefundStatus.filter(s => s.code !== currentRefundStatus && s.display)"
```
Comment (seeker25) [inline src/components/ViewRoutingSlip/RefundRequestForm.vue:84]: hmm, might be hard to debug this

File: src/components/ViewRoutingSlip/RefundRequestForm.vue [hunk 2]
```diff
@@ -82,24 +159,59 @@ import { RefundRequestDetails } from '@/models/RoutingSlip'
...
       chequeAdviceRules,
+      currentRefundStatus,
       name,
       chequeAdvice,
       address,
       addressForm,
       addressValidity,
       isValid,
-      canEdit
+      canEdit,
```
Comment (seeker25) [inline src/components/ViewRoutingSlip/RefundRequestForm.vue:199]: state, toRefs pattern?

File: src/composables/useRoutingSlip.ts [hunk 3]
```diff
@@ -175,6 +175,44 @@ export const useRoutingSlip = () => {
     }
   }
 
+  const updateRoutingSlipRefundStatus = async (
+    status: any
+  ) => {
+    const slipNumber = routingSlip.value.number
+    try {
+      const response = await RoutingSlipService.updateRoutingSlipRefundStatus(status, slipNumber)
```
Comment (seeker25) [inline src/composables/useRoutingSlip.ts:183]: would refactor into the service, a composable doesn't care if the status is 200, 203, it just cares if it gets the data or not

Comment (seeker25) [reply src/composables/useRoutingSlip.ts:183]: Also think about if you were to use this call in another component, you'd be copy and pasting alot of the response handling code etc

File: src/composables/useRoutingSlip.ts [hunk 4]
```diff
@@ -175,6 +175,44 @@ export const useRoutingSlip = () => {
...
+      if (response?.data && (response.status === 200 || response.status === 202)) {
```
Comment (seeker25) [inline src/composables/useRoutingSlip.ts:184]: Also what happens if the response doesn't have data? but doesn't error?

File: src/services/routingSlip.services.ts [hunk 5]
```diff
@@ -52,6 +52,24 @@ export default class RoutingSlip {
     )
   }
 
+  public static async updateRoutingSlipRefundStatus (
+    code: string,
+    routingSlipNumber: string
+  ): Promise<AxiosResponse> {
```
Comment (seeker25) [inline src/services/routingSlip.services.ts:58]: What type is the response? if it's simple, wouldn't recommend an interface.. if it's more complex and you use it, would recommend it

**Comment by seeker25** [comment]
You'll have to build library

File: src/services/routingSlip.services.ts (hunk 5 above)
Comment (Jxio) [reply src/services/routingSlip.services.ts:58]: it's same as updating status before.
image

File: tests/unit/components/ViewRoutingSlip/RoutingSlipInfo.spec.ts [hunk 6]
```diff
@@ -139,18 +139,25 @@ describe('RoutingSlipInfo.vue', () => {
...
+      },
       directives: {
         can () { /* stub */ }
-      },
-      mocks: { i18n }
+      }
     })
 
     await wrapper.vm.$nextTick()
+    console.log(wrapper.html())
```
Comment (seeker25) [inline tests/unit/components/ViewRoutingSlip/RoutingSlipInfo.spec.ts:160]: probably should remove

File: src/services/routingSlip.services.ts [hunk 7]
```diff
@@ -52,6 +52,44 @@ export default class RoutingSlip {
...
+    try {
+      const response = await axios.post(
+        `${ConfigHelper.getFasAPIURL()}/routing-slips/${slipNumber}/comments`,
+        data
+      )
+      if (response?.data) {
+        return response.data
+      }
+      return null
```
Comment (seeker25) [inline src/services/routingSlip.services.ts:86]: throw error here? instead of returning null

**Review by seeker25 — APPROVED** [review]

**Pull request merged**